src/operator/tensor/elemwise_unary_op.h

```diff
@@ -451,7 +451,7 @@ void CastCompute(const nnvm::NodeAttrs& attrs,
   Stream<xpu> *s = ctx.get_stream<xpu>();
   MSHADOW_TYPE_SWITCH_WITH_BOOL(outputs[0].type_flag_, DstDType, {
     Tensor<xpu, 1, DstDType> out = outputs[0].FlatTo1D<xpu, DstDType>(s);
-    MSHADOW_TYPE_SWITCH(inputs[0].type_flag_, SrcDType, {
+    MSHADOW_TYPE_SWITCH_WITH_BOOL(inputs[0].type_flag_, SrcDType, {
```

Comment:
Please reorganize `test_np_ndarray_astype` so that this case would be covered. Ideally, we should do
```python
dtypes = [np.int8, np.uint8, np.int32, np.float16, np.float32, np.float64,  np.bool, np.bool_,
          'int8', 'uint8', 'int32', 'float16', 'float32', 'float64', 'bool']
for itype, otype in itertools.product(dtypes, dtypes):
    verify_astype(itype, otype)
```

Reply:
Done.